presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/ColumnCardinalityCache.java

```diff
@@ -60,37 +81,43 @@
     private final Authorizations auths;
     private static final Logger LOG = Logger.get(ColumnCardinalityCache.class);
     private final Connector connector;
-    private final int size;
-    private final Duration expireDuration;
-
-    @GuardedBy("this")
-    private final Map<String, TableColumnCache> tableToCache = new HashMap<>();
+    private final ExecutorService executorService;
+    private final LoadingCache<CacheKey, Long> cache;
 
+    @SuppressWarnings("unchecked")
     public ColumnCardinalityCache(
             Connector connector,
             AccumuloConfig config,
             Authorizations auths)
     {
         this.connector = requireNonNull(connector, "connector is null");
-        this.size = requireNonNull(config, "config is null").getCardinalityCacheSize();
-        this.expireDuration = config.getCardinalityCacheExpiration();
         this.auths = requireNonNull(auths, "auths is null");
-    }
-
-    /**
-     * Deletes any cache for the given table, no-op of table does not exist in the cache
-     *
-     * @param schema Schema name
-     * @param table Table name
-     */
-    public synchronized void deleteCache(String schema, String table)
-    {
-        LOG.debug("Deleting cache for %s.%s", schema, table);
-        if (tableToCache.containsKey(table)) {
-            // clear the cache and remove it
-            getTableCache(schema, table).clear();
-            tableToCache.remove(table);
+        int size = requireNonNull(config, "config is null").getCardinalityCacheSize();
+        Duration expireDuration = config.getCardinalityCacheExpiration();
+
+        // Create executor service with one hot thread, pool size capped at 4x processors,
```

Comment:
Got it, I'll take the `BoundedExecutor` route and use injection and `@PreDestroy` to shutdown the executor.